tonic/src/transport/server/mod.rs

```diff
@@ -173,6 +173,10 @@ impl<L> Server<L> {
     /// builder.concurrency_limit_per_connection(32);
     /// ```
     #[must_use]
+    #[deprecated(
+        since = "0.12.4",
+        note = "Use `layer()` with a layer which provide concurrency limit feature such as `tower::limit::ConcurrencyLimitLayer` used in this config."
```

Comment:
I don't think that will provide 'per connection' concurrency-limiting semantics like this purports to.

Reply:
Makes sense. Seems necessary to add functionality to set layers for the purpose.

Reply:
I think that would be helpful, see e.g. https://github.com/hyperium/tonic/pull/1616